src/clients/BridgeClient.ts

```diff
@@ -254,7 +255,7 @@ export class BridgeClient implements IClient {
             this._logger.trace(TelemetryEvent.BridgeClient_ConnectSuccess);
         }
         catch (error) {
-            this._logger.error(TelemetryEvent.BridgeClient_ConnectError, error);
+            this._logger.error(TelemetryEvent.BridgeClient_ConnectError, redactJsonObject(error));
```

Comment:
when we do the long term work item of re-evaluating logging we should also re-evaluate this catch and throw design. throwing and catching is very expensive. If possible I would preffer to get away from this design patter.

Reply:
Indeed, Yes totally, as a design we should have `contracts` so the invalid json should never happen, because that violates the baseline of forming json messages if that is the baseline of the `error` contract, but for now, this get us over the line of having `secrets` which is privacy gap we should fill in and come out with more clear 3 steps process like:

Long term:

* First identify what we need in `TelemetryEvents` Pm might help us here.
* Second, bulk delete all which are not used and underwent linkable and linked information.
* Third, YAGNI - if the Invalid Json will never we supplied, we don't need to handle that case at all. 

Thanks.

Reply:
Agree with most, but this comment wasnt around the catch on the json redact :) This catch/throw is not even around invalid Json, its just catching the connect error from cli, logging and throwing again instead of just return false and some error to the caller. So even after invalid json is gone, i think we should still fix this pattern. ConnectAsync should return false and error message if connect fails. Definitely not for this PR, but wanted to note it so we can add work item for it.

Reply:
It will be two separate work items 1. Re-evaluate the logs and keep the one's needed with help of PM's (refactor the telemetry logs). 2. Refactor try/catch logic for different methods like connectAsync, prepConnect, refreshCreds etc.. return false and useful message to customer/user if that operation fails. correct ? It is better to have distinct work items for these just to have smaller and cleaner PR's is my opinion. Thanks.

Reply:
Sounds good, I will leave that for atleast you guys to discuss this offline, whatever you guys think is right portion moving forward, I would highly recommend if we drop whatever logs we don’t need as priority, 

* all events reside here: https://github.com/Azure/vscode-bridge-to-kubernetes/blob/main/src/logger/TelemetryEvent.ts

Thanks.